src/bp/ui-studio/src/web/views/OneFlow/diagram/index.tsx

```diff
@@ -736,14 +739,27 @@ class Diagram extends Component<Props> {
       newContents[index][this.state.currentLang] = data
     }
 
+    this.setState({ editingNodeItem: { node: { ...node, contents: newContents }, index } })
+
     this.props.updateFlowNode({ contents: newContents })
   }
 
+  updateNodeCondition(data) {
+    const { node, index } = this.state.editingNodeItem
+    const newConditions = [...node.conditions]
+
+    newConditions[index] = data
```

Comment:
if you intent to modify the copied array, this will modify the original array as conditions are object and are passed by reference to the newConditions arr..

if this has effects, you might want to change this logic for a pure implementation
```const conditions = [...node.conditions.slice(0, idx), data, ...node.conditions.slice(idx+1)]```

Reply:
using the spread operator creates a copy of the array, not a deep one, but it doesn't matter in this instead since I overwrite the object completely 🤔  it doesn't even really matter since the state in React is immutable, I guess I just do it this way by habbit